packages/@aws-cdk/aws-cognito-identitypool/test/identitypool.test.ts

```diff
@@ -0,0 +1,539 @@
+import {
+  Template,
+} from '@aws-cdk/assertions';
+import {
+  UserPool,
+  UserPoolIdentityProvider,
+} from '@aws-cdk/aws-cognito';
+import {
+  Role,
+  ServicePrincipal,
+  ArnPrincipal,
+  AnyPrincipal,
+  OpenIdConnectProvider,
+  SamlProvider,
+  SamlMetadataDocument,
+  PolicyStatement,
+  Effect,
+  PolicyDocument,
+} from '@aws-cdk/aws-iam';
+import {
+  Stack,
+} from '@aws-cdk/core';
+import {
+  IdentityPool,
+  IdentityPoolProviderUrl,
+} from '../lib/identitypool';
+import {
+  RoleMappingMatchType,
+} from '../lib/identitypool-role-attachment';
+import { UserPoolAuthenticationProvider } from '../lib/identitypool-user-pool-authentication-provider';
+
+describe('identity pool', () => {
+  test('minimal setup', () => {
+    const stack = new Stack();
+    new IdentityPool(stack, 'TestIdentityPoolMinimal');
+    const temp = Template.fromStack(stack);
+
+    temp.hasResourceProperties('AWS::Cognito::IdentityPool', {
+      AllowUnauthenticatedIdentities: false,
+    });
+
+    temp.resourceCountIs('AWS::IAM::Role', 2);
+    temp.resourceCountIs('AWS::IAM::Policy', 0);
+    temp.hasResourceProperties('AWS::IAM::Role', {
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: 'sts:AssumeRoleWithWebIdentity',
+            Condition: {
+              'StringEquals': {
+                'cognito-identity.amazonaws.com:aud': {
+                  Ref: 'TestIdentityPoolMinimalCA44517F',
+                },
+              },
+              'ForAnyValue:StringLike': {
+                'cognito-identity.amazonaws.com:amr': 'authenticated',
+              },
+            },
+            Effect: 'Allow',
+            Principal: {
+              Federated: 'cognito-identity.amazonaws.com',
+            },
+          },
+        ],
+      },
+    });
+
+  });
+
+  test('providing default roles directly', () => {
```

Comment:
This test looks pretty similar to the next one. Should we assert the IAM role properties rather than the policies in this one?

Reply:
The major differences are in the policies, but I added a test for role properties too just to be thorogh.